crates/nargo_cli/src/panic.rs

```diff
@@ -0,0 +1,80 @@
+// based on this code: https://github.com/zkat/miette/blob/main/src/panic.rs
+
+use backtrace::Backtrace;
+use miette::{Diagnostic, WrapErr};
+use thiserror::Error;
+
+const PANIC_MESSAGE: &str = "This is a bug. We may have already fixed this in newer versions of Nargo so try searching for similar issues at https://github.com/noir-lang/noir/issues/.\nIf there isn't an open issue for this bug, consider opening one at https://github.com/noir-lang/noir/issues/new?labels=bug&template=bug_report.yml";
+
+pub(crate) fn set_hook() {
+    std::panic::set_hook(Box::new(move |info| {
+        let mut report: miette::Result<()> = Err(Panic.into());
+        if let Some(loc) = info.location() {
+            report = report
+                .with_context(|| format!("at {}:{}:{}", loc.file(), loc.line(), loc.column()));
+        }
+        if let Err(err) = report.with_context(|| PANIC_MESSAGE.to_string()) {
+            eprintln!("{err:?}");
+        }
+    }));
+}
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("{0}{}", Panic::backtrace())]
+#[diagnostic()]
+struct Panic;
+
+impl Panic {
+    fn backtrace() -> String {
+        use std::fmt::Write;
+        if let Ok(var) = std::env::var("RUST_BACKTRACE") {
+            if !var.is_empty() && var != "0" {
+                const HEX_WIDTH: usize = std::mem::size_of::<usize>() + 2;
+                // Padding for next lines after frame's address
+                const NEXT_SYMBOL_PADDING: usize = HEX_WIDTH + 6;
+                let mut backtrace = String::new();
+                let trace = Backtrace::new();
+                let frames = backtrace_ext::short_frames_strict(&trace).enumerate();
+                for (idx, (frame, sub_frames)) in frames {
+                    let ip = frame.ip();
```

Comment:
Can we switch back to color-eyre now that the miette experiment did not pan out? Looking at panic messages from both, I prefer the color-eyre formatting. It seems to require less code as well.

Reply:
we can only use color-eyre for panics. at least that way you don't have to fork color-eyre and eyre

Reply:
Why can't we remove miette entirely?

Reply:
it looks like I was wrong.